pkg/storage/storagewrappers/boundedconcurrency.go

```diff
@@ -0,0 +1,97 @@
+package storagewrappers
+
+import (
+	"context"
+	"time"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var _ storage.RelationshipTupleReader = (*boundedConcurrencyTupleReader)(nil)
+
+var (
+	timeWaitingCounter = promauto.NewHistogram(prometheus.HistogramOpts{
+		Name:    "time_waiting_for_read_queries",
+		Help:    "Time (in ms) spent waiting for Read, ReadUserTuple and ReadUsersetTuples calls to the datastore",
+		Buckets: []float64{1, 10, 25, 50, 100, 1000, 5000}, // milliseconds
```

Comment:
```suggestion
		Buckets: []float64{10, 25, 50, 100, 1000}, // milliseconds
```

Reply:
a request can take up to 5000 ms (config.UpstreamTimeout), so i'm inclined to leave that bucket